src/marketplace/ListingsView.js

```diff
@@ -0,0 +1,146 @@
+//@flow strict
+import * as React from 'react';
+import {getTankComponent, verifyComponent} from '../armory/GetInventoryInfo.js';
+import Cookies from 'universal-cookie';
+import type {ComponentType} from '../armory/ComponentType.js';
+
+type Props = {|
+	//This is the type of item we are buying
+	sellerType: ComponentType,
+|};
+type State = {|
+	userId: string,
+|};
+
+//This allows for all the items that are for sale to be with in one array
+let itemsForSale = [{name: '', price: 0, amount: 0, sellerId: '', saleId: ''}];
+class ListingsView extends React.Component<Props, State> {
+	constructor(props:Props) {
+		super(props);
+		this.state={
+			userId: '',
+		}
+	}
+
+	//This is the initial functions that we call
+	componentDidMount() {
+		this.getUserId();
+		this.getMarketSales();
+	}
+
+	//When sellerType is updated we need to get the new sells
+	componentDidUpdate(prevProps:Props) {
+		if (prevProps.sellerType !== this.props.sellerType) {
+			this.getMarketSales();
+		}
+	}
+
+	//This gets us the user's id 
+	getUserId = async ():Promise<void> => {
+		const cookies = new Cookies();
+		const token = cookies.get('token');
+		const response = await fetch('/api/user/getUser/', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+				'x-auth-token': token
+			},
+		});
+		const jsonObjectOfUser = await response.json();
+		this.setState({userId:jsonObjectOfUser._id});
+	};
+
+	//Gets all the sells and filters them based on what type we are currently looking at
+	getMarketSales = async ():Promise<void> => {
+		itemsForSale = [];
+		const response = await fetch('/api/marketplace/getMarketSales/', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+			},
+		});
+		const jsonObjectOfSells = await response.json();
+		for (const sale in jsonObjectOfSells)
+		{
+			const typeOfItem = getTankComponent(verifyComponent(jsonObjectOfSells[sale].itemId));
+			if(typeOfItem === this.props.sellerType)
+			{
+				//This is a temp obj used to create an object of what a sell item would need to have. It is pushed into an array of all of the items for sell that fall in the current seller type
+				let obj = {};
+				obj['name'] = jsonObjectOfSells[sale].itemId;
+				obj['price'] = jsonObjectOfSells[sale].salePrice;
+				obj['amount'] = jsonObjectOfSells[sale].amount;
+				obj['sellerId'] = jsonObjectOfSells[sale].sellerId;
+				obj['saleId'] = jsonObjectOfSells[sale]._id;
+				itemsForSale.push(obj);
+			}
+		}
+		//Need this to deal with the asynch nature of api calling......fun times
+		//Makes sure that we create the cards after we get the sells
+		this.forceUpdate();     
```

Comment:
We shouldn't be doing this. This is indicating that you are using react incorrectly